### ISSUE TITLE
Belajar AI bareng Dicoding X Microsoft Elevate Training Center

### ISSUE THREAD

**Issue by roiskhoiron** [roiskhoiron/life.dotdev#25]
### Discussed in https://github.com/roiskhoiron/.github/discussions/24

<div type='discussions-op-text'>

<sup>Originally posted by **roiskhoiron** April 26, 2026</sup>
Saya dapet info kursus gratis dari dicoding dan microsoft tentang AI dari sini https://dicoding.elevaite.id</div>

**Comment by roiskhoiron** [comment]
> ### Discussed in [#24](https://github.com/roiskhoiron/.github/discussions/24)
> 
> Originally posted by **roiskhoiron** April 26, 2026 Saya dapet info kursus gratis dari dicoding dan microsoft tentang AI dari sini https://dicoding.elevaite.id

Materi ini diadopsi dari platform microsoft learning ini https://learn.microsoft.com/en-us/training/paths/implement-data-science-machine-learning-fabric/

**Comment by roiskhoiron** [comment]
# Catatan Belajar: Analitik End-to-End dengan Microsoft Fabric

**Gaya: ringkas, santai, plus diagram**

---

## 📌 Latar Belakang (Dulu tuh ribet)

```mermaid
flowchart TD
    subgraph Tradisional [Alur Kerja Tradisional]
        DE[Data Engineer<br/>ambil data mentah sendiri] -->|koordinasi jelek| DA[Data Analyst<br/>dapet data kurang konteks bisnis]
        DA -->|analisis terganggu| DS[Data Scientist<br/>susah integrasi model native]
        DS -->|infra kompleks| Problem[❌ Proyek molor, biaya besar]
    end
```

**Hambatan yang muncul:**
- Koordinasi kurang baik saat pengambilan data mentah
- Analyst ikut terdampak karena data kehilangan konteks bisnis
- Data scientist kesulitan integrasi teknik native (Python, R, MLflow)
- Infrastruktur kompleks, nggak efisien, nggak cepat

---

## 🚀 Solusi: Microsoft Fabric dengan Workspace OneLake

```mermaid
flowchart LR
    subgraph Fabric [Microsoft Fabric - Kolaboratif]
        OneLake[OneLake<br/>Penyimpanan terpusat di atas ADLS] --> Workspace[Workspace<br/>Wadah proyek semua tim]
        Workspace --> Lakehouse[Lakehouse<br/>Tempat kelola, cleaning, analisis]
        Lakehouse --> Tim[Data Engineer<br/>Data Analyst<br/>Data Scientist]
        Tim --> |satu data, satu konteks| Hasil[✅ Cepat, aman, terintegrasi]
    end
    
    Admin[API/SDK Admin<br/>Otomatisasi & kontrol akses] -.-> Workspace
```

**Fitur andalan:**
- **OneLake** – penyimpanan terpusat, hilangkan silo
- **Lakehouse** – kerja bareng kelola + cleaning + analisis
- **Workspace** – proyek nyatu semua artefak
- **Kontrol akses** – RBAC, aman tapi kolaboratif
- **Format populer** – Delta, Parquet, CSV, JSON
- **API/SDK** – admin bisa otomatisasi semuanya

---

## 📊 Perbandingan Langsung

```mermaid
graph TB
    subgraph Sebelum[Sebelum: Tradisional]
        A1[Data Engineer] -->|manual transfer| A2[Data Analyst]
        A2 -->|data sudah berubah| A3[Data Scientist]
        A3 -->|integrasi sulit| A4[Lambat & Mahal]
    end
    
    subgraph Sesudah[Sesudah: Fabric]
        B1[Data Engineer] --> B4[OneLake & Lakehouse]
        B2[Data Analyst] --> B4
        B3[Data Scientist] --> B4
        B4 -->|satu sumber data<br/>konteks utuh| B5[Cepat & Efisien]
    end
```

| Aspek | Tradisional (pusing) | Fabric Workspace OneLake (enak) |
|-------|----------------------|----------------------------------|
| Koordinasi | Serial, sering miss | Real-time, semua tim lihat data sama |
| Dampak ke analyst | Data nggak sesuai konteks | Bisa akses langsung & perbaiki di Lakehouse |
| Konteks bisnis | Sering hilang | Terjaga karena sumber data tunggal |
| Integrasi data science | Ribet, perlu migrasi | Native Python/MLflow di atas Delta/Parquet |
| Infrastruktur | Kompleks, terpisah | Terpusat, API/SDK, elastis |
| Kecepatan proyek | Minggu–bulan | Hari–jam |

---

## 🧠 Takeaway

1. **Silo data musuh utama** → diatasi OneLake
2. **Lakehouse = pusat operasi** → semua tim kerja bareng
3. **Kontrol akses penting** → jangan sampai kolaborasi mengorbankan keamanan
4. **Format terbuka (Delta, Parquet)** → data scientist happy
5. **API/SDK admin** → infrastruktur as code, admin tenang
6. **Alur kolaboratif Fabric** → eliminasi semua hambatan tradisional

---

## 🔑 Hashtag

`#MicrosoftFabric` `#OneLake` `#Lakehouse` `#EndToEndAnalytics` `#DataSiloElimination` `#DataEngineer` `#DataAnalyst` `#DataScientist` `#CollaborativeWorkflow` `#KonteksBisnis` `#NativeDataScience` `#DeltaLake` `#Parquet` `#RBAC` `#APIdanSDK` `#InfrastructureAsCode` `#FabricAdminPortal` `#BelajarData` `#DearybookCatatan`

---

## 📝 Penutup

Hari ini kita udah belajar dari **alur tradisional yang ruwet** menuju **Workspace OneLake di Microsoft Fabric** yang kolaboratif, cepat, dan aman. Diagram di atas bisa langsung kamu pakai di catatan Markdown-mu.

**Comment by roiskhoiron** [comment]
---

# 📓 Rangkuman Utuh: Dunia Data Science dengan Microsoft Fabric

**Hari belajar:** 1  
**Metode:** Pemaparan konsep + panduan praktik lab  
**Status:** Siap review & praktik

---

## 🧠 1. Apa Itu Data Science (Terapan)?

> **Data science** adalah pekerjaan mencari **pola-pola kompleks** dari aktivitas analisis data, dan kita dapat memanfaatkannya untuk menciptakan **model bagi AI**.

Bukan sekadar statistik biasa, tapi pola rumit yang menjadi "otak" AI.

`#DataScience` `#PolaKompleks` `#ModelAI`

---

## 🧠 2. Melatih Model Bukan Satu-satunya Tugas

Data scientist dalam satu proyek harus melakukan:
1. Memahami masalah bisnis
2. Mengumpulkan data
3. Membersihkan & prep data
4. Eksplorasi data (EDA)
5. Feature engineering
6. Melatih model
7. Evaluasi & tuning
8. Deployment & monitoring
9. Maintenance & retraining

**Melatih model hanyalah satu langkah dari banyak langkah.**

`#BukanCumaModel` `#SiklusHidupDataScience`

---

## 🧠 3. Jenis Model AI yang Umum Digunakan

| Jenis Model | Output | Contoh Algoritma |
|-------------|--------|------------------|
| **Klasifikasi** | Kategori (spam/non-spam) | Logistic Regression, Random Forest, XGBoost |
| **Regresi** | Nilai kontinu (harga rumah) | Linear Regression, Ridge, Lasso |
| **Clustering** | Kelompok tanpa label | K-Means, DBSCAN |
| **Forecasting** | Deret waktu (penjualan bulan depan) | ARIMA, Prophet, LSTM |

> **Prinsip:** Pilih model berdasarkan **masalah bisnis**, bukan karena tren.

`#Klasifikasi` `#Regresi` `#Clustering` `#Forecasting` `#BusinessProblemFirst`

---

## 🧠 4. 5 Langkah Berkesinambungan dalam Proyek Data Science

```mermaid
flowchart LR
    A[Mencari Permasalahan] --> B[Collecting Data]
    B --> C[Preparasi Data]
    C --> D[Explorasi Data]
    D --> E[Training Model]
    E --> F[We Got Insight]
    F -.-> A
```

| Langkah | Nama | Aktivitas |
|---------|------|-----------|
| 1 | Mencari permasalahan | Pahami bisnis, rumuskan pertanyaan data |
| 2 | Collecting data | Ambil data dari berbagai sumber |
| 3 | Data preparation | Bersihkan, handle missing, transformasi |
| 4 | Exploration data | EDA, statistik, visualisasi awal |
| 5 | Training model | Pilih algoritma, latih, evaluasi |
| + | Insight | Wawasan bisnis atau model siap pakai |

`#5LangkahDataScience` `#CRISPDM`

---

## 🧠 5. Preparasi Data: Tahap Paling Menguras Waktu & Paling Berpengaruh

> **80% waktu** dalam proyek data science dihabiskan untuk preparasi data.

**Aktivitas preparasi:**
- Handle missing values
- Deteksi & tangani outlier
- Normalisasi / standarisasi
- Encoding variabel kategorikal
- Feature extraction & engineering

**Prinsip:** Garbage in, garbage out. Semakin teliti preparasi, semakin baik model.

`#PreparasiData` `#DataCleaning` `#80PercentTime` `#GarbageInGarbageOut`

---

## 🧠 6. Volume + Kualitas Data = Performa Model

> Semakin banyak volume dataset pelatihan (dengan kualitas baik), semakin tinggi akurasi model.

| Dimensi | Pengaruh |
|---------|----------|
| **Volume besar** | Model belajar variasi pola, mengurangi overfitting |
| **Kualitas tinggi** | Data bersih, representatif, tidak bias, label akurat |

**Kombinasi volume besar + kualitas tinggi = model optimal.**

`#VolumeVsKualitas` `#AkurasiModel` `#DataBerkualitas`

---

## 🧠 7. Data Ingestion & Processing di Fabric: Code-First atau Low-Code

| Pendekatan | Tools | Cocok untuk |
|-------------|-------|--------------|
| **Low-code / No-code** | Dataflow Gen2, Data Pipeline (Copy Activity) | Business analyst, citizen developer |
| **Code-first** | Notebook (PySpark, Python, R, SQL), Apache Airflow Jobs | Data engineer, data scientist, developer |

**Keunggulan:** Satu platform untuk semua peran, tanpa silo data.

`#DataIngestion` `#CodeFirst` `#LowCode` `#DataflowGen2` `#FabricNotebook` `#ApacheAirflow`

---

## 🧠 8. Alur Praktis Menggunakan Fabric (Ingestion → Lakehouse → EDA → Wrangler)

```mermaid
flowchart TD
    Sumber[Cloud, CSV Lokal, ADLS] --> Ingestion[Dataflow/Pipeline]
    Ingestion --> Lakehouse[OneLake Lakehouse]
    Lakehouse --> Notebook[Notebook + Apache Spark]
    Notebook --> EDA[EDA & Transformasi]
    EDA --> Wrangler[Data Wrangler<br/>Ringkasan Statistik]
    Wrangler --> Cleaning[Data Cleaning Cepat]
```

**Detail:**
- **Ingestion:** Data terstruktur/tidak terstruktur dari cloud, CSV lokal, ADLS → masuk Lakehouse.
- **Processing:** Apache Spark di Notebook (PySpark, Python, R) untuk big data.
- **Data Wrangler:** Berikan ringkasan statistik (mean, median, missing values, distribusi) agar mudah ambil keputusan cleaning.
- **Visualisasi:** Banyak opsi visual di notebook untuk pemahaman awal.

`#AlurFabric` `#Ingestion` `#Lakehouse` `#ApacheSpark` `#DataWrangler` `#EDA`

---

## 🧠 9. Pelatihan Model Berulang & MLflow untuk Tracking

> Melatih model perlu dilakukan **berulang** (iterative) untuk mencari konfigurasi terbaik.  
> **MLflow di Fabric** memudahkan **monitoring setiap run** (parameter, metrik, artefak) sehingga kita bisa membandingkan dan memilih **model terbaik untuk production**.

```mermaid
flowchart LR
    Iterasi[Training Run 1..n] --> MLflow[Catat parameter, metrik, artefak]
    MLflow --> Compare[Bandingkan semua runs]
    Compare --> Best[Pilih model terbaik]
    Best --> Production[Deploy ke production]
```

**Scoring & Batch Prediction:**
- **Scoring:** Evaluasi model dengan data test → dapat metrik (akurasi, precision, dll).
- **Batch prediction:** Gunakan fungsi `predict()` untuk memprediksi banyak data baru sekaligus (contoh: prediksi penjualan minggu depan).

`#IterativeTraining` `#MLflow` `#ExperimentTracking` `#BatchPrediction` `#Scoring`

---

## 🧪 Panduan Praktik Lab (dari Microsoft Learn)

**Topik lab:** [Get started with data science in Microsoft Fabric](https://microsoftlearning.github.io/mslearn-fabric/Instructions/Labs/08-data-science-get-started.html)  
**Dataset:** Diabetes (Azure Open Datasets)  
**Tools:** Notebook, PySpark, Data Wrangler, MLflow, Experiments

**Langkah ringkas:**
1. **Buat Workspace** di Fabric (kapasitas Trial/Premium/Fabric)
2. **Buat Notebook** baru
3. **Muat dataset** diabetes menggunakan kode dari lab
4. **Eksplorasi dengan Data Wrangler** (statistik ringkas, visualisasi cepat)
5. **Lakukan persiapan data** (handle missing, transformasi)
6. **Latih dua model** (misal: Logistic Regression vs Random Forest)
7. **Tracking dengan MLflow** → buka item "Experiments" di workspace untuk bandingkan metrik
8. **(Opsional) Prediksi batch** untuk data baru

> **Tips:** Nyalakan Spark session terlebih dahulu, dan gunakan Data Wrangler dengan klik ikon di output DataFrame.

`#LabFabric` `#DataScienceLab` `#MLflowInAction` `#GetStartedFabric`

---

## 🔑 Daftar Hashtag Lengkap (untuk referensi & sosial)

`#MicrosoftFabric` `#DataScience` `#OneLake` `#Lakehouse` `#EndToEndAnalytics`  
`#PolaKompleks` `#ModelAI` `#JenisModel` `#Klasifikasi` `#Regresi` `#Clustering` `#Forecasting`  
`#5LangkahDataScience` `#PreparasiData` `#80PercentTime` `#DataCleaning`  
`#VolumeVsKualitas` `#DataIngestion` `#CodeFirst` `#LowCode` `#DataflowGen2` `#FabricNotebook`  
`#ApacheSpark` `#PySpark` `#DataWrangler` `#EDA` `#IterativeTraining` `#MLflow`  
`#ExperimentTracking` `#BatchPrediction` `#Scoring` `#LabFabric` `#GetStartedFabric`  
`#BelajarData` `#DearybookCatatan`

---

## 📝 Penutup

Hari ini kita sudah menempuh perjalanan panjang dari **fondasi konsep data science** hingga **panduan praktik lab di Microsoft Fabric**:
- Memahami apa itu data science dan bahwa melatih model bukan satu-satunya tugas.
- Mengenal 4 jenis model dan 5 langkah siklus proyek.
- Menyadari bahwa preparasi data adalah kunci utama (80% waktu, sangat berpengaruh).
- Memahami bahwa volume + kualitas data menentukan performa model.
- Mengetahui opsi ingestion & processing di Fabric (low-code dan code-first).
- Mengikuti alur praktis: ingestion → lakehouse → Spark notebook → Data Wrangler → cleaning.
- Belajar tentang iterative training dan kemudahan tracking dengan MLflow.
- Menyiapkan diri untuk praktik lab langsung dari Microsoft Learn.

**Besok kita lanjut ke praktik sesungguhnya:** menjalankan notebook di Fabric, melatih model dengan tracking MLflow Experiments, dan mencoba batch prediction.

Siapkan akun Fabric Trial Anda dan semangat! 😊

`#Bersambung` `#PraktikFabric` `#MLflowExperiments` `#SampaiJumpa`

---

**Comment by roiskhoiron** [comment]
<img width="1334" height="989" alt="Image" src="https://github.com/user-attachments/assets/15ac7faa-2952-4c0d-9183-af880b1244eb" />

**Comment by roiskhoiron** [comment]
![sertifikat_course_708_263610_040526155804.pdf](https://github.com/user-attachments/files/27356167/sertifikat_course_708_263610_040526155804.pdf)
